lobbyboy/conf/lobbyboy_config.toml

```diff
@@ -1,26 +1,27 @@
-data_dir = "/Users/xintao.lai/Programs/lobbyboy/dev_datadir2"
-listen_port = 12200
-listen_ip = "0.0.0.0"
-destroy_interval = "1m"
+data_dir = "./dev_datadir"
```

Comment:
yes, self explain is great and lobbyboy's config has been done well, user can understand each config in `lobbyboy_config.toml` without going to any other place.

however, i think there are a little more configs show to user in the file, and users may need to spend a little time to understand what are necessary config when they adding new config part(eg: for `provider`, `min_life_to_live` is necessary or not?).

i put the config control to python level. on the one hand, users don't need to care much about the config (meanwhile, they also can get explanatino of each config), on the other hand, lobbyboy can use config more flexibly (eg config validation or default for everyone or some simple preprocessing,  i think it's friendly to dev at least🤣).
and this will not affect the validity of the config file, any changes in config file will be fed back to python level, what would you think?

Reply:
> i  put the config control to python level.

It is good to have an python level representation of config.

My concern is that, for e.g. https://github.com/laixintao/lobbyboy/pull/16/files/1f03108637a39753ed3c817a27f5583ba57fed3e#diff-5106cb0e18f7aaa12a74a17f90a62c8509cf3fd8d035de87297b2922a97d337fR46

the `min_life_to_live` in comment config is `0s`, but in code it is `min_life_to_live: str = "50m"`. Even though we can change the config example, but I think it is better that we always read from config and don't have any default value in code, so that we will have a single of truth.

> however, i think there are a little more configs show to user in the file, and users may need to spend a little time to understand what are necessary config when they adding new config part(eg: for provider, min_life_to_live is necessary or not?).

I think that won't be a problem, every provider should have a config example in documentation, and user should read this before using. Also, we have a default value for every config, so he/she doesn't have to worry, the config example is for them to copy, then change it if necessary.

Reply:
make sense, let's keep single of truth rule, i'll remove default value from python level

Reply:
thanks